### PR TITLE
[ADD] port lettermgmt

### PR DESCRIPTION
**Pull request by hbrunn**

### CODE REVIEW

File: lettermgmt/models/__init__.py [hunk 1]
```diff
@@ -18,12 +17,10 @@
...
-###############################################################################
-
+##############################################################################
 from . import letter_class
 from . import letter_folder
 from . import res_letter
 from . import letter_type
 from . import letter_reassignment
-from . import letter_history
 from . import letter_channel
```
Comment (ehdem) [inline lettermgmt/models/__init__.py]: Better to do:

from . import (
    letter_class,
    letter_folder,
    res_letter,
    letter_type,
    letter_reassignment,
    letter_channel,
)

File: lettermgmt/models/letter_folder.py [hunk 2]
```diff
@@ -19,20 +19,19 @@
...
         'name': fields.char('Name', required=True),
-        'code': fields.char('Code', size=8, required=True),
-        'letter_ids': fields.one2many('res.letter', 'folder_id', string='Letters',
-                                      help='Letters contained in this folder.'),
+        'code': fields.char('Code', required=True),
+        'letter_ids': fields.one2many(
+            'res.letter', 'folder_id', string='Letters',
+            help='Letters contained in this folder.'),
     }
     _sql_constraints = [('code_uniq', 'unique(code)', 'Code must be unique !')]
```
Comment (ehdem) [inline lettermgmt/models/letter_folder.py:37]: Put _('Code must be unique !') to be translated

File: lettermgmt/models/letter_type.py [hunk 3]
```diff
@@ -20,17 +20,16 @@
...
     _name = 'letter.type'
-    _description = _("Letter Type")
+    _description = "Letter Type"
     _columns = {
-        'name': fields.char('Type', size=32, required=True),
-        'code': fields.char('Code', size=8, required=True),
+        'name': fields.char('Type', required=True),
+        'code': fields.char('Code', required=True),
     }
     _sql_constraints = [('code_uniq', 'unique(code)', 'Code must be unique !')]
```
Comment (ehdem) [inline lettermgmt/models/letter_type.py:35]: Put _('Code must be unique !') to be translated

File: lettermgmt/models/__init__.py (hunk 1 above)
Comment (guewen) [inline lettermgmt/models/__init__.py]: I disagree. When importing several things from the same module, it is a usual practice to group them, but import of  different modules is used to be on several lines.

File: lettermgmt/models/letter_folder.py (hunk 2 above)
Comment (guewen) [inline lettermgmt/models/letter_folder.py:37]: AFAIK this is not necessary on sql constraints.

**Comment by hbrunn** [comment]
@guewen @ehdem okay, grouped imports it is
@ehdem constraint messages are translated automatically

**Comment by hbrunn** [comment]
also here, I merged #15

File: __unported__/lettermgmt/letter_history.py [hunk 4]
```diff
@@ -1,41 +0,0 @@
...
-import time
-
-
-class letter_history(orm.Model):
-    _name = "letter.history"
-    _description = _("Letter Communication History")
-    _order = "id desc"
-    _columns = {
-        'register_id': fields.many2one('res.letter', 'Register'),
-        'name': fields.char('Action', size=64),
```
Comment (StefanRijnhart) [inline __unported__/lettermgmt/letter_history.py:35]: Moving the functionality to chatter seems like a good idea. Can you post any existing history items as chatter messages by means of migration?

File: lettermgmt/models/res_letter.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            [('in', 'IN'), ('out', 'OUT')], 'Move', readonly=True,
+            help="Incoming or Outgoing Letter."),
+        'type': fields.many2one(
+            'letter.type', 'Type',
+            help="Type of Letter, Depending upon size."),
+        'class': fields.many2one(
+            'letter.class', 'Class', help="Classification of Document."),
+        'date': fields.datetime('Letter Date', help='The letter\'s date'),
+        'snd_rec_date': fields.datetime(
+            'Sent / Received Date', required=True,
```
Comment (StefanRijnhart) [inline lettermgmt/models/res_letter.py]: This field is required now. I understand that the date received of an incoming letter can be required as it should be known in any state. But in the case of an outgoing letter, should it maybe be kept without a default and be set when setting the state to 'Sent' if it is still empty at that point?

File: __unported__/lettermgmt/letter_history.py (hunk 4 above)
Comment (hbrunn) [inline __unported__/lettermgmt/letter_history.py:35]: This code was already dead in the 7.0 version, so a migration would have to take place there

File: lettermgmt/models/res_letter.py (hunk 5 above)
Comment (hbrunn) [inline lettermgmt/models/res_letter.py]: done

**Comment by StefanRijnhart** [comment]
Thanks! :+1:

**Comment by pedrobaeza** [comment]
I would put as module name only 'letter', as mgmt, management and so on seems redundant. As this can be generalized to other documents, maybe another name is also advisable, but that implies a lot of changes in all labels.

File: lettermgmt/__openerp__.py [hunk 6]
```diff
@@ -22,44 +22,30 @@
 
 {
     'name': 'Letter Management',
-    'version': '0.1',
+    'version': '0.2',
```
Comment (pedrobaeza) [inline lettermgmt/__openerp__.py]: Why not 1.0?

File: lettermgmt/__openerp__.py [hunk 7]
```diff
@@ -22,44 +22,30 @@
...
     'author': 'Savoir-faire Linux',
     'maintainer': 'Savoir-faire Linux',
     'website': 'http://www.savoirfairelinux.com',
     'license': 'AGPL-3',
     'category': 'Social Network',
```
Comment (pedrobaeza) [inline lettermgmt/__openerp__.py]: This category doesn't fit exactly, doesn't it?

**Comment by hbrunn** [comment]
@pedrobaeza mind you that this a port of an existing module, so I wouldn't rename it if not absolutely necessary

File: lettermgmt/__openerp__.py (hunk 7 above)
Comment (hbrunn) [inline lettermgmt/__openerp__.py]: 'Customer Relationship Management' better?

File: lettermgmt/__openerp__.py (hunk 6 above)
Comment (hbrunn) [inline lettermgmt/__openerp__.py]: does it matter?

Comment (hbrunn) [inline lettermgmt/__openerp__.py]: it makes sense to use 1.2 here as the code for v7 is version 1.1

File: lettermgmt/__openerp__.py (hunk 7 above)
Comment (pedrobaeza) [inline lettermgmt/__openerp__.py]: Yeah!

**Comment by pedrobaeza** [comment]
This is the moment for the renaming, as this is the entry PR to this version, but as you want.

**Comment by StefanRijnhart** [comment]
@pedrobaeza Renaming standalone community modules is expensive for its users. While we can cover the migration easily in OpenUpgrade, this is not the case for upgrades covered by the Odoo enterprise contract. Non-expert users would be left with inconsistent databases and be forced to perform an uninstall/fresh install cycle of the module, losing pre-existing data in the process. So while I agree with your claim that the current name is not optimal, I would still vote against renaming it.

**Comment by pedrobaeza** [comment]
OK, as you prefer. I suggested that because the number of DBs that is already using this in v7 that we can't cover and that is going to migrate to v8 is minimal.

:+1:

**Pull request merged**